src/spatialdata/_core/operations/map.py

```diff
@@ -0,0 +1,122 @@
+        The function to apply to the data.
+    fn_kwargs
+        Additional keyword arguments to pass to the function `func`.
+    blockwise
+        If `True`, distributed processing will be achieved with `dask.array.map_overlap`/`dask.array.map_blocks`,
+        otherwise the function is applied to the full data. If `False`, `depth` and `chunks` are ignored.
+    depth
+        If not `None`, distributed processing will be achieved with `dask.array.map_overlap`, otherwise with
+        `dask.array.map_blocks`. Specifies the overlap between chunks, i.e. the number of elements that each chunk
+        should share with its neighbor chunks. Please see `dask.array.map_overlap` for more information on the accepted
+        values.
+    chunks
+        Passed to `dask.array.map_overlap`/`dask.array.map_blocks` as `chunks`. Ignored if `blockwise` is `False`.
+        Chunk shape of resulting blocks if the function does not preserve the data shape. If not provided, the resulting
+        array is assumed to have the same chunk structure as the first input array.
+        E.g. ( (3,), (100,100), (100,100) ).
+    c_coords
+        The channel coordinates for the output data. If not provided, the channel coordinates of the input data are
+        used. It should be specified if the function changes the number of channels.
+    dims
+        The dimensions of the output data. If not provided, the dimensions of the input data are used. It must be
+        specified if the function changes the data dimensions.
+        E.g. ('c', 'y', 'x').
+    transformations
+        The transformations of the output data. If not provided, the transformations of the input data are copied to the
+        output data. It should be specified if the function changes the data transformations.
+    kwargs
+        Additional keyword arguments to pass to `dask.array.map_overlap` or `dask.array.map_blocks`.
+
+    Returns
+    -------
+    The processed data as a `DataArray`.
+    """
+    if isinstance(data, DataArray):
+        arr = data.data
+    elif isinstance(data, DataTree):
+        arr = data["scale0"].values().__iter__().__next__().data
+    else:
+        raise ValueError("Only 'DataArray' and 'DataTree' are supported.")
+
+    model = get_model(data)
+    if model in (Labels2DModel, Labels3DModel) and c_coords is not None:
+        raise ValueError("Channel coordinates can not be provided for labels data.")
+
+    kwargs = kwargs.copy()
+    kwargs["chunks"] = chunks
+
+    if not blockwise:
+        arr = func(arr, **fn_kwargs)
+    else:
+        if depth is not None:
+            kwargs.setdefault("boundary", "reflect")
+
+            if not isinstance(depth, int) and len(depth) != arr.ndim:
+                raise ValueError(
+                    f"Depth {depth} is provided for {len(depth)} dimensions. "
+                    f"Please (only) provide depth for {arr.ndim} dimensions."
+                )
+            kwargs["depth"] = coerce_depth(arr.ndim, depth)
```

Comment:
If `depth` would be passed in `kwargs`, it will appear as an argument, not in `kwargs`, so in that respect I do not see a problem. 
Instead of putting `depth` and `chunks` in `kwargs`, we could indeed also do the following:

`map_func(func, arr, chunks=chunks, depth=depth,  **fn_kwargs, **kwargs, dtype=arr.dtype)`

We could also remove `chunks` and `depth` from the list of parameters altogether, and simply let users pass them via kwargs. In that scenario, I would also remove these sanity checks on `depth` and delegate them to `da.map_overlap`.

Re `coerce_depth`. We have three options:
1. Remove coerce_depth from our code, and let users pass whatever map_overlap can work with, i.e. we delegate conversion to a dict to map_overlap (which will also call coerce_depth).
2. Remove coerce_depth, and only accept e.g. tuples; and add some sanity checks. But then we are a bit too restrictive I think, escpecially for users that are used to work with `map_overlap`and prefer to pass e.g. a dict.
3. Keep the code as is. Having coerce_depth in our code, makes it a bit more explicit as what is exactly passed to map_overlap, which could help debugging later on.

I have a preferance for 3, but not too strong of an opinion on it